### ISSUE TITLE
Import-Module bccontainerhelper fails if executed inside a container where docker is not installed

### ISSUE THREAD

**Issue by makuhn** [microsoft/navcontainerhelper#2571]
We are using a slim docker container image for compiling the apps via build pipeline. For that we create corresponding docker images that include the relevant Business Central apps as well as AL compiler. These components are downloaded using bccontainerhelper. A modification introduced with version 3.0.11 now causes the following error when creating the compiler image:

*The term 'docker' is not recognized as the name of a cmdlet*

This error is caused by the statement `Import-Module bccontainerhelper -DisableNameChecking`. Docker container is created using the following dockerfile (I only added the part that is relevant for raising this error):

```
# escape=`
ARG BASEVERSION
ARG BASETYPE
FROM mcr.microsoft.com/windows/servercore:$BASEVERSION as mother
LABEL yaveoncompilerbuilder=true
ARG NCHVERSION
SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop';"]

RUN Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force; `
    Install-Module bccontainerhelper -MinimumVersion $env:NCHVERSION -MaximumVersion $env:NCHVERSION -Force;

ARG BCTYPE
ARG BCCOUNTRY
ARG BCVERSION
ARG BCSTORAGEACCOUNT
ARG BCSASTOKEN

RUN Import-Module bccontainerhelper -DisableNameChecking; `
    Download-Artifacts -artifactUrl (Get-BCArtifactUrl -type $env:BCTYPE -country $env:BCCOUNTRY -version $env:BCVERSION -storageAccount $env:BCSTORAGEACCOUNT -sasToken $env:BCSASTOKEN) -includePlatform;

...
```

The error is caused by following code in BcContainerHelper.psm1 (line:151 char:26) - see also attachedlog from build pipeline:

```
$myinspect = docker inspect $(hostname) |  ...
```

Can we expect that bccontainerhelper is modified to support this scenario? Otherwise we must extend our compiler image.

[compiler.log](https://github.com/microsoft/navcontainerhelper/files/9011237/compiler.log)

**Comment by freddydk** [comment]
Will fix this

**Comment by freddydk** [comment]
Just added a try/catch around the docker command - preview build should be available in ~40 min.

**Comment by freddydk** [comment]
fixed in the latest preview

**Comment by freddydk** [comment]
Shipped in 4.0.3
